Empty Tuple with Any Type Results in Unused Variable

I added the following test case to [run.test](https://github.com/JukkaL/mypyc/blob/04eeafbaf2721a2614b0e7f061c02ada5ab20905/test-data/run.test):
```
[case testEmptyTupleFunctionWithAnyType]
from typing import Any
def f() -> Any:
    return ()
[file driver.py]
from native import f
assert f() == ()
```

Running `pytest -k testEmptyTupleFunctionWithAnyType` results in the following error:
```
============================== test session starts ===============================
platform darwin -- Python 3.6.5, pytest-3.6.3, py-1.5.4, pluggy-0.6.0
rootdir: /Users/ed1d1a8d/dev/mypyc, inifile: pytest.ini
plugins: xdist-1.22.2, forked-0.2, cov-2.5.1
collected 332 items / 331 deselected

mypyc/test/test_run.py F                                                   [100%]

==================================== FAILURES ====================================
_______________________ testEmptyTupleFunctionWithAnyType ________________________
data: /Users/ed1d1a8d/dev/mypyc/test-data/run.test:210:
/Users/ed1d1a8d/dev/mypyc/mypyc/buildc.py:134: in run_setup_py_build
    subprocess.check_output(['python', setup_path, 'build'], stderr=subprocess.STDOUT)
/Users/ed1d1a8d/.pyenv/versions/3.6.5/lib/python3.6/subprocess.py:336: in check_output
    **kwargs).stdout
/Users/ed1d1a8d/.pyenv/versions/3.6.5/lib/python3.6/subprocess.py:418: in run
    output=stdout, stderr=stderr)
E   subprocess.CalledProcessError: Command '['python', './setup.py', 'build']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:
/Users/ed1d1a8d/dev/mypyc/mypyc/test/test_run.py:108: in run_case
    native_lib_path = buildc.build_c_extension(cpath, mod, preserve_setup=True)
/Users/ed1d1a8d/dev/mypyc/mypyc/buildc.py:27: in build_c_extension
    return run_setup_py_build(setup_path, module_name)
/Users/ed1d1a8d/dev/mypyc/mypyc/buildc.py:136: in run_setup_py_build
    raise BuildError(err.output)
E   mypyc.buildc.BuildError: Build failed
------------------------------ Captured stdout call ------------------------------
==================== Generated C ====================
0    #include <Python.h>
1    #include <CPy.h>
2
3    static CPyModule *CPyStatic_module;
4    static PyObject *CPyStatic_globals;
5    static CPyModule *CPyStatic_builtins_module;
6    struct tuple_def_821329250005071 {
7    };
8
9    static PyObject *CPyDef_f(void);
10   static PyObject *CPyPy_f(PyObject *self, PyObject *args, PyObject *kw);
11   static PyObject *CPyDef___top_level__(void);
12
13   static PyMethodDef module_methods[] = {
14       {"f", (PyCFunction)CPyPy_f, METH_VARARGS | METH_KEYWORDS, NULL /* docstring */},
15       {NULL, NULL, 0, NULL}
16   };
17
18   static struct PyModuleDef module = {
19       PyModuleDef_HEAD_INIT,
20       "native",
21       NULL, /* docstring */
22       -1,       /* size of per-interpreter state of the module,
23                    or -1 if the module keeps state in global variables. */
24       module_methods
25   };
26
27   PyMODINIT_FUNC PyInit_native(void)
28   {
29       if (CPyStatic_module != NULL) {
30           Py_INCREF(CPyStatic_module);
31           return CPyStatic_module;
32       }
33       CPyStatic_module = PyModule_Create(&module);
34       if (CPyStatic_module == NULL)
35           return NULL;
36       CPyStatic_globals = PyModule_GetDict(CPyStatic_module);
37       if (CPyStatic_globals == NULL)
38           return NULL;
39       if (CPyStatic_builtins_module == NULL) {
40           CPyStatic_builtins_module = PyImport_ImportModule("builtins");
41           if (CPyStatic_builtins_module == NULL)
42               return NULL;
43       }
44       CPyModule *CPyStatic_typing_module;
45       CPyStatic_typing_module = PyImport_ImportModule("typing");
46       if (CPyStatic_typing_module == NULL)
47           return NULL;
48       PyObject *__tmp1 = PyObject_GetAttrString(CPyStatic_typing_module, "Any");
49       if (__tmp1 == NULL)
50           return NULL;
51       if (PyModule_AddObject(CPyStatic_module, "Any", __tmp1) < 0)
52          return NULL;
53       Py_DECREF(CPyStatic_typing_module);
54       PyObject *result = CPyDef___top_level__();
55       if (result == NULL)
56           return NULL;
57       Py_DECREF(result);
58       return CPyStatic_module;
59   }
60
61   static PyObject *CPyDef_f(void) {
62       struct tuple_def_821329250005071 cpy_r_r0;
63       PyObject *cpy_r_r1;
64   CPyL0: ;
65       cpy_r_r1 = PyTuple_New(0);
66       if (cpy_r_r1 == NULL)
67           CPyError_OutOfMemory();
68       return cpy_r_r1;
69   }
70
71   static PyObject *CPyPy_f(PyObject *self, PyObject *args, PyObject *kw) {
72       static char *kwlist[] = {0};
73       if (!PyArg_ParseTupleAndKeywords(args, kw, ":f", kwlist)) {
74           return NULL;
75       }
76       return CPyDef_f();
77   }
78
79   static PyObject *CPyDef___top_level__(void) {
80       PyObject *cpy_r_r0;
81   CPyL0: ;
82       cpy_r_r0 = Py_None;
83       Py_INCREF(cpy_r_r0);
84       return cpy_r_r0;
85   }
==================== End C ====================
==================== Build output ====================
running build
running build_ext
building 'native' extension
creating build
creating build/temp.macosx-10.12-x86_64-3.6
creating build/temp.macosx-10.12-x86_64-3.6/tmp
clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/ed1d1a8d/dev/mypyc/mypyc/../lib-rt -I/Users/ed1d1a8d/.pyenv/versions/mypyc/include -I/Users/ed1d1a8d/.pyenv/versions/3.6.5/include/python3.6m -c tmp/native.c -o build/temp.macosx-10.12-x86_64-3.6/tmp/native.o -Wno-unused-function -Wno-unused-label -Werror -Wno-unreachable-code
tmp/native.c:63:38: error: unused variable 'cpy_r_r0' [-Werror,-Wunused-variable]
    struct tuple_def_821329250005071 cpy_r_r0;
                                     ^
1 error generated.
error: command 'clang' failed with exit status 1
==================== End output ====================
==================== 1 failed, 331 deselected in 0.60 seconds ====================
```

--- a/mypyc/buildc.py
+++ b/mypyc/buildc.py
@@ -87,7 +87,7 @@ import sys
 module = Extension('{package_name}',
                    sources=['{cpath}'],
                    extra_compile_args=['-Wno-unused-function', '-Wno-unused-label', '-Werror',
-                                       '-Wno-unreachable-code'],
+                                       '-Wno-unreachable-code', '-Wno-unused-variables'],
                    libraries=[{libraries}],
                    library_dirs=[{library_dirs}])
 

